protobuf-kafkaconnect-converter/src/main/java/com/amazonaws/services/schemaregistry/kafkaconnect/protobuf/fromconnectdata/ConnectDataToProtobufDataConverterFactory.java

```diff
@@ -0,0 +1,18 @@
+package com.amazonaws.services.schemaregistry.kafkaconnect.protobuf.fromconnectdata;
+
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+import org.apache.kafka.connect.data.Schema;
+
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class ConnectDataToProtobufDataConverterFactory {
+    public static DataConverter get(final Schema connectSchema) {
+        final Schema.Type connectType = connectSchema.type();
+
+        if (connectType.isPrimitive()) {
+            return new PrimitiveDataConverter();
+        }
+
+        throw new IllegalArgumentException("Unrecognized connect type: " + connectType);
```

Comment:
Do we need a TODO here for non-primitive types

Reply:
it's under development. We. have a project plan to track it.